src/Common/ErrorCodes.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <stddef.h>
+#include <cstdint>
+#include <utility>
+#include <atomic>
+#include <common/types.h>
+#include <string_view>
+
+/** Allows to count number of simultaneously happening error codes.
+  * See also Exception.cpp for incrementing part.
+  */
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    /// ErrorCode identifier (index in array).
+    using ErrorCode = size_t;
```

Comment:
It looks strange.

Reply:
This was done by analogy from the CurrentMetrics.h and similar